relay-server/src/actors/global_config.rs

```diff
@@ -257,23 +261,43 @@ impl Service for GlobalConfigService {
             let mut shutdown_handle = Controller::shutdown_handle();
 
             relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
-                // NOTE(iker): if this first request fails it's possible the default
-                // global config is forwarded. This is not ideal, but we accept it
-                // for now.
-                self.update_global_config();
-            } else {
-                // NOTE(iker): not making a request results in the sleep handler
-                // not being reset, so no new requests are made.
-                relay_log::info!("fetching global configs disabled: no credentials configured");
-            }
+            match self.config.relay_mode() {
+                RelayMode::Managed => {
+                    if self.config.has_credentials() {
+                        relay_log::info!("serving global configs fetched from upstream");
+                        self.request_global_config();
+                    } else {
+                        relay_log::info!(
+                            "serving default global configs due to lacking credentials"
+                        );
+                    }
+                }
+                RelayMode::Static | RelayMode::Proxy | RelayMode::Capture => {
+                    match GlobalConfig::load(self.config.path()) {
+                        Ok(Some(from_file)) => {
+                            relay_log::info!("serving static global config loaded from file");
+                            self.global_config_watch.send(Arc::new(from_file)).ok();
+                        }
+                        Ok(None) => {
+                            relay_log::info!(
+                                "serving default global configs due to lacking static global config file"
+                            );
+                        }
+                        Err(e) => {
+                            relay_log::error!("failed to load global config from file: {}", e);
+                            relay_log::info!(
+                                "serving default global configs due to failure to load global config from file"
+                            );
+                        }
+                    }
+                }
+            };
 
             loop {
                 tokio::select! {
                     biased;
 
-                    () = &mut self.fetch_handle => self.update_global_config(),
+                    () = &mut self.fetch_handle => self.request_global_config(),
```

Comment:
In the static mode we actually never get this `fetch_handle` trigger? Is it right?